### ISSUE TITLE
classRef/@except processing fails

### ISSUE THREAD

**Issue by martindholmes** [TEIC/Stylesheets#454]
The Stylesheets Working Group, trying to integrate the latest P5 changes into the Stylesheets repo, has discovered that classRef/@except is correctly processed for RNG/RNC, but for DTDs it creates an invalid DTD.

This is triggered by the change here:

https://github.com/TEIC/TEI/commit/193e6283321aeeb68b9a93b0d95b10a82148ef74

which closes ticket https://github.com/TEIC/TEI/issues/1848.

**Comment by lb42** [comment]
This looks the same as a problem reported in jan 2017 (#220)
 That ticket also includes a useful testfile.

**Comment by sydb** [comment]
Thanks, @lb42. Helpful test file. However, I do not think this is the same problem, though clearly related. 

This problem seems to be rooted, I have discovered (after several hours), in the CLASSMEMBERS key, defined in odds/teiodds. That key only matches `<classSpec>` elements, not `<classRef>`. I have fixed it (and CLASSMEMBERS-CLASSES, which has the same problem), and that has fixed the immediate problem. However, testing what else might go wrong will take quite a while. Once tested somewhat I will push up a new branch, **sydb-454** or some such.

**Comment by hcayless** [comment]
When this is fixed, the content models of `<lem>` and `<rdg>` will need to be updated.

**Comment by ebeshero** [comment]
Is this related to #754 ?

**Comment by sydb** [comment]
> Is this related to [#754](https://github.com/TEIC/Stylesheets/issues/754) ?

I do not think so, but not at all sure. It is, however, probably related to the failure of the obvious solution to TEIC/TEI#2460 — just adding `except="address"` to either `<classRef key="model.phrase">` or `<classRef key="model.divBottom">` seems to cause the entire class reference to be ignored.
